src/TorchSharp/NN/Normalization/NormBase.cs

```diff
@@ -0,0 +1,145 @@
+                    init.zeros_(this._running_mean);
+                    init.ones_(this._running_var);
+                    init.zeros_(this._num_batches_tracked);
+                }
+            }
+
+            public void reset_parameters() {
+                ResetRunningStats();
+                if (affine) {
+                    init.ones_(this._weight);
+                    init.zeros_(this._bias);
+                }
+            }
+
+            protected abstract void ValidateInputDimensions(Tensor input);
+
+            protected override void Dispose(bool disposing)
+            {
+                _weight?.Dispose();
+                _bias?.Dispose();
+                base.Dispose(disposing);
+            }
+
+            public Parameter? bias {
+                get => _bias;
+                set {
+                    _bias?.Dispose();
+                    _bias = value?.DetachFromDisposeScope() as Parameter;
+                    ConditionallyRegisterParameter(nameof(bias), _bias);
+                }
+            }
+
+            public Parameter weight {
+                get => _weight!;
+                set {
+                    if (value is null) throw new ArgumentNullException(nameof(weight));
+                    if (value.Handle != _weight?.Handle) {
+                        _weight?.Dispose();
+                        _weight = (value.DetachFromDisposeScope() as Parameter)!;
+                        ConditionallyRegisterParameter(nameof(weight), _weight);
+                    }
+                }
+            }
+
+            public Tensor? running_mean {
+                get => _running_mean;
+                set {
+                    _running_mean?.Dispose();
+                    _running_mean = value?.DetachFromDisposeScope();
+                    ConditionallyRegisterBuffer(nameof(running_mean), _running_mean);
+                }
+            }
+
+            public Tensor? running_var {
+                get => _running_var;
+                set {
+                    _running_var?.Dispose();
+                    _running_var = value?.DetachFromDisposeScope();
+                    ConditionallyRegisterBuffer(nameof(running_var), _running_var);
+                }
+            }
+
+            public Tensor? num_batches_tracked {
+                get => _num_batches_tracked;
+                set {
+                    _num_batches_tracked?.Dispose();
+                    _num_batches_tracked = value?.DetachFromDisposeScope();
+                    ConditionallyRegisterBuffer(nameof(num_batches_tracked), _num_batches_tracked);
+                }
+            }
+            
+            public long num_features { get; private set; }
+            
+            public double eps { get; private set; }
+            
+            public double momentum { get; private set; }
+
+            public bool affine { get; private set; }
+            
+            public bool track_running_stats { get; private set; }
```

Comment:
Why can't `eps`, `momentum`, be modified?

Reply:
Fixed